spec/statelint_spec.rb

```diff
@@ -422,4 +422,37 @@
     problems = linter.validate(j)
     expect(problems.size).to eq(1)
   end
+
+  it 'should allow ItemProcessor in Map' do
+    j = File.read("test/map-with-itemprocessor.json")
+    linter = StateMachineLint::Linter.new
+    problems = linter.validate(j)
+    expect(problems.size).to eq(0)
+  end
+
+  it 'should reject Map state with both Iterator and ItemProcessor' do
+    j = File.read("test/map-with-itemprocessor-and-iterator.json")
+    linter = StateMachineLint::Linter.new
+    problems = linter.validate(j)
+    expect(problems.size).to eq(1)
+  end
+
+  it 'should reject Map state with both Parameters and ItemSelector' do
+    j = File.read("test/map-with-parameters-and-itemselector.json")
+    linter = StateMachineLint::Linter.new
+    problems = linter.validate(j)
+    expect(problems.size).to eq(1)
+  end
+
+  it 'should allow valid TaskCredentials in Task' do
+    j = File.read "test/task-with-credentials.json"
+    linter = StateMachineLint::Linter.new
+    problems = linter.validate(j)
+    expect(problems.size).to eq(0)
+
+    j = File.read "test/task-with-invalid-credentials.json"
```

Comment:
This one should be a separate test case: "should reject Task state with invalid Credentials"